tracing-subscriber/src/reload.rs

```diff
@@ -36,7 +36,34 @@ pub struct Layer<L, S> {
     _s: PhantomData<fn(S)>,
 }
 
-/// Allows reloading the state of an associated `Layer`.
+/// Allows reloading the state of an associated [`Layer`](crate::layer::Layer).
+///
+/// # Examples
+///
+/// Reloading a [`Filtered`](crate::filter::Filtered) layer to change the filter at runtime.
+///
+/// ```
+/// # use tracing::info;
+/// # use tracing_subscriber::{filter,fmt,reload,Registry,prelude::*};
+/// # fn main() {
+/// let (filtered_layer, reload_handle) =
+///     reload::Layer::new(fmt::Layer::default().with_filter(filter::LevelFilter::WARN));
```

Comment:
nit, take it or leave it: i might split this out, like
```suggestion
/// let filtered_layer = fmt::Layer::default()
///     .with_filter(filter::LevelFilter::WARN);
/// let (filtered_layer, reload_handle) = reload::Layer::new(fiiltered_layer);
```